spec/integration/federation/federation_messages_generation.rb

```diff
@@ -0,0 +1,178 @@
+    root_guid:        @local_target.guid,
+    diaspora_id:      @remote_user.diaspora_handle,
+    public:           true
+  )
+
+  generate_xml(@entity, @remote_user, @user)
+end
+
+def mock_private_key_for_user(user)
+  expect(DiasporaFederation.callbacks).to receive(:trigger)
+                                            .with(:fetch_private_key_by_diaspora_id, user.person.diaspora_handle)
+                                            .once
+                                            .and_return(user.encryption_key)
+end
+
+def retraction_mock_callbacks(entity, sender)
+  return unless [
+    DiasporaFederation::Entities::SignedRetraction,
+    DiasporaFederation::Entities::RelayableRetraction
+  ].include?(entity.class)
+
+  mock_private_key_for_user(sender)
+
+  allow(DiasporaFederation.callbacks).to receive(:trigger)
+                                            .with(
+                                              :fetch_entity_author_id_by_guid,
+                                              entity.target_type,
+                                              entity.target_guid
+                                            )
+                                            .once
+                                            .and_return(sender.encryption_key)
+end
+
+def generate_retraction(entity_name, target_object, sender=@remote_user)
+  @entity = FactoryGirl.build(
+    entity_name,
+    diaspora_id: sender.diaspora_handle,
+    target_guid: target_object.guid,
+    target_type: target_object.class.to_s
+  )
+
+  retraction_mock_callbacks(@entity, sender)
+
+  generate_xml(@entity, sender, @user)
+end
+
+def generate_forged_retraction(entity_name, target_object, sender=@remote_user)
+  times = 1
+  if %i(signed_retraction_entity relayable_retraction_entity).include?(entity_name)
+    times += 2
+  end
+
+  substitute_wrong_key(sender, times)
+  generate_retraction(entity_name, target_object, sender)
+end
+
+def generate_relayable_local_parent(entity_name)
+  klass = FactoryGirl.factory_by_name(entity_name).build_class
+  @entity = FactoryGirl.build(
+    entity_name,
+    conversation_guid: @local_target.guid,
+    parent_guid:       @local_target.guid,
+    diaspora_id:       @remote_user.person.diaspora_handle,
+    poll_answer_guid:  @local_target.respond_to?(:poll_answers) ? @local_target.poll_answers.first.guid : nil
+  )
+
+  mock_private_key_for_user(@remote_user)
+
+  expect(DiasporaFederation.callbacks).to receive(:trigger)
+                                            .with(
+                                              :fetch_author_private_key_by_entity_guid,
+                                              klass.get_target_entity_type(@entity.to_h),
+                                              kind_of(String)
+                                            )
+                                            .and_return(nil)
+  generate_xml(@entity, @remote_user, @user)
+end
+
+def generate_relayable_remote_parent(entity_name)
```

Comment:
Assignment Branch Condition size for generate_relayable_remote_parent is too high. [24.1/20]